java-manta-client/src/main/java/com/joyent/manta/util/DigestCloner.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2016-2017, Joyent, Inc. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.joyent.manta.util;
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.bouncycastle.crypto.Digest;
+import org.bouncycastle.util.Memoable;
+
+/**
+ * Utility class for cloning Digest objects.
+ */
+public final class DigestCloner implements Cloner<Digest> {
+
+    @Override
+    public Digest createClone(final Digest source) {
+        if (source instanceof Memoable) {
+            return (Digest) ((Memoable) source).copy();
+        }
+
+        throw new NotImplementedException("Clone not implemented for type: " + source.getClass().getCanonicalName());
```

Comment:
Did you forget to finish this or is this intentional. Maybe a comment may be useful.